packages/components/src/components/form/Select/index.tsx

```diff
@@ -361,16 +361,18 @@ export const Select = ({
     );
 
     const Option = useCallback(
-        (optionProps: OptionProps<Option>) => (
+        (optionProps: OptionProps<Option, boolean>) => (
             <ReactSelectComponents.Option
                 {...optionProps}
                 innerProps={
-                    dataTest && optionProps.data.value
-                        ? ({
-                              ...optionProps.innerProps,
-                              'data-test': `${dataTest}/option/${optionProps.data.value}`,
-                          } as OptionProps<Option>['innerProps'])
-                        : optionProps.innerProps
+                    {
+                        ...optionProps.innerProps,
+                        'data-test': `${dataTest}/option/${
+                            typeof optionProps.data.value === 'string'
+                                ? optionProps.data.value
+                                : optionProps.label
```

Comment:
Ok, why not. I would prefer more to use only `label` for example but we would need to be sure that label is always present. Also it would require changing this in multiple other tests so feel free to ignore this comment